### PR TITLE
Add missing siteurl on single host

### LINKED ISSUE #27: Exception when generating sitemap when the current website has only 1 host set up
**Steps:** 
1. a Optimizely CMS 12.1.0 project with 1 host set up as below: 
image
2. Go To set up Seo Sitemaps, choose Standard site map, click Update/save
3. Run the Scheduled job to generate sitemaps
4. Error: Job successfully executed.<br/>Generated sitemaps: <br/>Error creating sitemap for "sitemap.xml" appear in History of the Scheduled job, with an exception in Backend: `fail: Geta.Optimizely.Sitemaps.XML.SitemapXmlGenerator[0]
      Error on generating xml sitemap
      System.ArgumentNullException: Value cannot be null. (Parameter 'uriString')
         at System.Uri..ctor(String uriString)
         at Geta.Optimizely.Sitemaps.XML.SitemapXmlGenerator.Generate(SitemapData sitemapData, Boolean persistData, Int32& entryCount)
`
5. Sitemap is not generated and it's not accessible

**Workaround:**
Have a website with multiple hosts, that way when we go to Seo Sitemaps admin, there's a dropdown to choose hosts, then click Update, the SiteUrl is not null anymore. After this, Generating sitemaps work as normal and things work fine.

I think this is a bug because when there is no dropdown to choose the Host, the SiteUrl is not passed into the Api call, and thus the exception occurs.

### PR DESCRIPTION
**Pull request by brianweet**
Fixes #27

### CODE REVIEW

File: src/Geta.Optimizely.Sitemaps/Pages/Geta.Optimizely.Sitemaps/Index.cshtml [hunk 1]
```diff
@@ -35,14 +35,15 @@
...
-                                    <label>@Model.HostLabel</label>
-                                }
                                 @if (Model.ShowHostsDropDown)
                                 {
                                     <select asp-for="SitemapViewModel.SiteUrl" asp-items="Model.SiteHosts"></select>
                                 }
+                                else
+                                {
+                                    <label>@Model.HostLabel</label>
+                                    <input type="hidden" class="form-control align-middle" asp-for="SitemapViewModel.SiteUrl" value="@Model.HostLabel"/>
```
Comment (marisks) [inline src/Geta.Optimizely.Sitemaps/Pages/Geta.Optimizely.Sitemaps/Index.cshtml:45]: Do you need CSS class for the hidden input?

File: src/Geta.Optimizely.Sitemaps/Pages/Geta.Optimizely.Sitemaps/Index.cshtml [hunk 2]
```diff
@@ -136,14 +137,15 @@
...
-                                <label asp-for="SitemapViewModel.RelativePath"></label>
-                            }
                             @if (Model.ShowHostsDropDown)
                             {
                                 <select asp-for="SitemapViewModel.SiteUrl" asp-items="Model.SiteHosts"></select>
                             }
+                            else
+                            {
+                                <label asp-for="SitemapViewModel.RelativePath"></label>
+                                <input type="hidden" class="form-control align-middle" asp-for="SitemapViewModel.SiteUrl" value="@Model.HostLabel"/>
```
Comment (marisks) [inline src/Geta.Optimizely.Sitemaps/Pages/Geta.Optimizely.Sitemaps/Index.cshtml:147]: Do you need CSS class for the hidden input?

Comment (marisks) [reply src/Geta.Optimizely.Sitemaps/Pages/Geta.Optimizely.Sitemaps/Index.cshtml:147]: Is `asp-for="SitemapViewModel.SiteUrl"` correct here? Shouldn't it be `asp-for="SitemapViewModel.RelativePath"`?

**Review by marisks — COMMENTED** [review]
In general, I see that `@Model.ShowHostsLabel` was always `true` and it always rendered a label - also when a dropdown was visible. Don't we need a label for dropdowns?

File: src/Geta.Optimizely.Sitemaps/Pages/Geta.Optimizely.Sitemaps/Index.cshtml (hunk 1 above)
Comment (brianweet) [reply src/Geta.Optimizely.Sitemaps/Pages/Geta.Optimizely.Sitemaps/Index.cshtml:45]: Haha nope, will remove it

File: src/Geta.Optimizely.Sitemaps/Pages/Geta.Optimizely.Sitemaps/Index.cshtml (hunk 2 above)
Comment (brianweet) [reply src/Geta.Optimizely.Sitemaps/Pages/Geta.Optimizely.Sitemaps/Index.cshtml:147]: It is correct, the issue only existed if the host dropdown wasn't shown - when there's only 1 host.

**Comment by brianweet** [comment]
ShowHostsLabel was only set to `true` if there was just 1 host (I removed the part from an `if .. else` statement)

**Comment by brianweet** [comment]
And regarding the label in the UI. We should just clean it up completely, the UI is quite messy at the moment. I will create a new issue for that.

**Pull request merged**